### PR TITLE
Missing required options

### PR DESCRIPTION
**Pull request by windoze**
* Missing required options when using AccessToken to read AzureSQL.
* Support SQL query in addition to DBTable.

### CODE REVIEW

**Review by xiaoyongzhu — APPROVED** [review]

File: src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
   override def loadDf(ss: SparkSession, dataIOParameters: Map[String, String] = Map()): DataFrame = {
     var reader = ss.read.format("jdbc")
       .option("url", url)
     if (StringUtil.isBlank(dbtable)) {
       // Fallback to default table name
       reader = reader.option("dbtable", ss.conf.get(DBTABLE_CONF))
     } else {
-      reader = reader.option("dbtable", dbtable)
+      val q = dbtable.trim
+      if("\\s".r.findFirstIn(q).nonEmpty) {
```
Comment (jaymo001) [inline src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala:20]: I might prefer we add this in a separate PR. Is it needed for now? Is it supported by all JDBC sources? Can we add some test cases?

File: src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
+import com.fasterxml.jackson.annotation.JsonAlias
 import com.fasterxml.jackson.module.caseclass.annotation.CaseClassDeserialize
 import com.linkedin.feathr.offline.source.dataloader.jdbc.JdbcUtils
 import com.linkedin.feathr.offline.source.dataloader.jdbc.JdbcUtils.DBTABLE_CONF
 import org.apache.spark.sql.{DataFrame, SparkSession}
 import org.eclipse.jetty.util.StringUtil
 
 @CaseClassDeserialize()
-case class Jdbc(url: String, dbtable: String, user: String = "", password: String = "", token: String = "", useToken: Boolean = false, anonymous: Boolean = false) extends InputLocation {
+case class Jdbc(url: String, @JsonAlias(Array("query")) dbtable: String, user: String = "", password: String = "", token: String = "", useToken: Boolean = false, anonymous: Boolean = false) extends InputLocation {
```
Comment (hangfei) [inline src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala:11]: documentation

File: src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
+        reader = reader.option("query", q)
+      } else {
+        reader = reader.option("dbtable", q)
+      }
     }
     if (useToken) {
-      reader.option("accessToken", LocationUtils.envSubstitute(token)).load
+      reader.option("accessToken", LocationUtils.envSubstitute(token))
+        .option("hostNameInCertificate", "*.database.windows.net")
+        .option("encrypt", true)
```
Comment (hangfei) [inline src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala:30]: document why it's always true

File: src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala [hunk 4]
```diff
@@ -1,24 +1,34 @@
...
+        // This is a SQL instead of a table name
+        reader = reader.option("query", q)
+      } else {
+        reader = reader.option("dbtable", q)
+      }
     }
     if (useToken) {
-      reader.option("accessToken", LocationUtils.envSubstitute(token)).load
+      reader.option("accessToken", LocationUtils.envSubstitute(token))
+        .option("hostNameInCertificate", "*.database.windows.net")
```
Comment (hangfei) [inline src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala:29]: document why is this set this way for future dev.

File: src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala (hunk 3 above)
Comment (windoze) [reply src/main/scala/com/linkedin/feathr/offline/config/location/Jdbc.scala:30]: These properties are required by AzureSQL Spark connector, they're set according to Azure docs at https://docs.microsoft.com/en-us/azure/azure-sql/database/spark-connector#connecting-using-an-access-token

**Comment by xiaoyongzhu** [comment]
@windoze can you merge main into your branch to trigger the CI test?

**Review by xiaoyongzhu — APPROVED** [review]

**Review by jaymo001 — APPROVED** [review]

**Pull request merged**